File: implementation/scipamato/scipamato-common-utils/src/test/java/ch/difty/scipamato/common/DateUtilsTest.java
package ch.difty.scipamato.common;

import static org.assertj.core.api.Assertions.assertThat;

import java.sql.Timestamp;

import org.junit.jupiter.api.Test;

class DateUtilsTest extends FinalClassTest<DateUtils> {

    @Test
    void tsOf_withNullDate() {
        assertThat(DateUtils.tsOf(null)).isNull();
    }

    @Test
    void tsOf_withNonNullDate() {
        Timestamp ts = Timestamp.valueOf("2017-01-03 13:55:12.123");
        assertThat(DateUtils.tsOf(ts.toLocalDateTime())).isEqualTo(ts);
    }

}

File: implementation/scipamato/common/utils/src/test/kotlin/ch/difty/scipamato/common/DateUtilsTest.kt
package ch.difty.scipamato.common

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.sql.Timestamp

internal class DateUtilsTest : FinalClassTest<DateUtils> {

    @Test
    fun tsOf_withNullDate() {
        assertThat(DateUtils.tsOf(null)).isNull()
    }

    @Test
    fun tsOf_withNonNullDate() {
        val ts = Timestamp.valueOf("2017-01-03 13:55:12.123")
        assertThat(DateUtils.tsOf(ts.toLocalDateTime())).isEqualTo(ts)
    }

}
